weather/Weather.Viz.py

```diff
@@ -1,6 +1,7 @@
 from matplotlib import  dates, pyplot as plt
 import pandas as pd
 from api_reader import getYearData
+import matplotlib.pyplot as plt
```

Comment:
Duplicate import: Matplotlib is already imported above, you can remove this duplicate import.